.ci/atime/tests.R

```diff
@@ -118,7 +118,17 @@ test.list <- atime::atime_test_list(
       data.table:::setattr(L, "class", NULL)
       data.table:::setDT(L)
     },
-    Slow = "c4a2085e35689a108d67dacb2f8261e4964d7e12", # Parent of the first commit in the PR that fixes the issue (https://github.com/Rdatatable/data.table/commit/7cc4da4c1c8e568f655ab5167922dcdb75953801)
-    Fast = "1872f473b20fdcddc5c1b35d79fe9229cd9a1d15") # Last commit in the PR that fixes the issue (https://github.com/Rdatatable/data.table/pull/5427/commits)
+    Slow = "c4a2085e35689a108d67dacb2f8261e4964d7e12",  # Parent of the first commit in the PR that fixes the issue (https://github.com/Rdatatable/data.table/commit/7cc4da4c1c8e568f655ab5167922dcdb75953801)
+    Fast = "1872f473b20fdcddc5c1b35d79fe9229cd9a1d15"), # Last commit in the PR that fixes the issue (https://github.com/Rdatatable/data.table/pull/5427/commits)
+  "fread(colClasses=Date) improved in #6107" = atime::atime_test(
+    N = 10^seq(1, 7),
+    setup = {
+      DT = data.table(date=.Date(sample(20000, N, replace=TRUE))),
+      tmp_csv = tempfile()
+      fwrite(DT, tmp_csv)
+    },
+    expr = fread(tmp_csv, colClasses=list(Date='date')),
```

Comment:
cc @Anirban166 @tdhock one consideration here -- actually there are 3 independent codepaths (at present) for potential regression:

 1. As here, with `colClasses = list(Date = ...)`
 2. With "anonymous" columns, i.e. `fread(f, colClasses="Date")`, required when column names are not known in advance
 3. With `select=` (see description in `?fread`)

I think ideally we'd combine some of the logic, but for now, it looks like we'd want 3 separate regression tests for each of these, WDYT?

Also, I am wondering what happens to `tmp_csv` -- when is it cleaned up? I don't see any `cleanup=` step in `atime_versions()`, do we need to wait for the R session to end for those (potentially large) files to be removed?

Reply:
> I think ideally we'd combine some of the logic, but for now, it looks like we'd want 3 separate regression tests for each of these, WDYT?

I agree (three separate regression tests would be better for now)

> Also, I am wondering what happens to `tmp_csv` -- when is it cleaned up? I don't see any `cleanup=` step in `atime_versions()`, do we need to wait for the R session to end for those (potentially large) files to be removed?

I'm not sure about the cleanup (Toby would know better), but I'm wondering too if we can have something like a `file.remove` or `unlink` call on `tmp_csv`

Reply:
> I think ideally we'd combine some of the logic, but for now, it looks like we'd want 3 separate regression tests for each of these, WDYT?

I agree (three separate regression tests would be better for now)

> Also, I am wondering what happens to `tmp_csv` -- when is it cleaned up? I don't see any `cleanup=` step in `atime_versions()`, do we need to wait for the R session to end for those (potentially large) files to be removed?

I'm not sure about the cleanup (Toby would know better), but I'm wondering too if we can have something like a `file.remove` or `unlink` call on `tmp_csv` as well

Reply:
> I think ideally we'd combine some of the logic, but for now, it looks like we'd want 3 separate regression tests for each of these, WDYT?

I agree (three separate regression tests would be better for now)

> Also, I am wondering what happens to `tmp_csv` -- when is it cleaned up? I don't see any `cleanup=` step in `atime_versions()`, do we need to wait for the R session to end for those (potentially large) files to be removed?

I'm not sure about the cleanup (Toby would know better), but I'm wondering too if we can have something like a `file.remove` or `unlink` call on `tmp_csv`

Reply:
right three separate tests.
yes we need to wait for the R session to end for those files to be removed.
they probably won't be too large because of the 0.01 second time limit.
also please make sure to do data.table::fread instead of just fread inside expr.